# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpfe/appshell/nsAppShellService.cpp
# Commit: aea4783f8b48
# Full Hash: aea4783f8b4867167607f8ba00eadc4aaa45d61d
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-02-07 03:55:51
# Regressor Bug: 1582832
# File Overlap Count: 1
# Description:
#   Bug 1582832: Part 3 - Correctly handle windowless BCs created for printing. r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D61006
# ==============================================================================

diff -r cefac6461948 -r aea4783f8b48 xpfe/appshell/nsAppShellService.cpp
--- a/xpfe/appshell/nsAppShellService.cpp	Thu Feb 06 19:08:04 2020 +0000
+++ b/xpfe/appshell/nsAppShellService.cpp	Thu Feb 06 19:08:06 2020 +0000
@@ -450,10 +450,10 @@
   NS_ENSURE_SUCCESS(rv, rv);
 
   // Create a BrowsingContext for our windowless browser.
-  RefPtr<BrowsingContext> browsingContext =
-      BrowsingContext::Create(nullptr, nullptr, EmptyString(),
-                              aIsChrome ? BrowsingContext::Type::Chrome
-                                        : BrowsingContext::Type::Content);
+  RefPtr<BrowsingContext> browsingContext = BrowsingContext::CreateWindowless(
+      nullptr, nullptr, EmptyString(),
+      aIsChrome ? BrowsingContext::Type::Chrome
+                : BrowsingContext::Type::Content);
 
   /* Next, we create an instance of nsWebBrowser. Instances of this class have
    * an associated doc shell, which is what we're interested in.
